neqo-qpack/src/encoder.rs

```diff
@@ -195,65 +197,48 @@ impl QPackEncoder {
         }
     }
 
-    #[cfg(test)]
-    fn insert_with_name_ref(
-        &mut self,
-        name_static_table: bool,
-        index: u64,
-        value: &[u8],
-    ) -> Res<()> {
-        qdebug!(
-            [self],
-            "insert with name reference {} from {} value={:x?}.",
-            index,
-            if name_static_table {
-                "static table"
-            } else {
-                "dynamic table"
-            },
-            value
-        );
-        self.table
-            .insert_with_name_ref(name_static_table, index, value)?;
-
-        // write instruction
-        if name_static_table {
-            EncoderInstruction::InsertWithNameRefStatic {
-                index,
-                value: value.to_vec(),
-            }
-            .marshal(&mut self.send_buf, self.use_huffman);
-        } else {
-            EncoderInstruction::InsertWithNameRefDynamic {
-                index,
-                value: value.to_vec(),
-            }
-            .marshal(&mut self.send_buf, self.use_huffman);
+    pub fn insert(&mut self, conn: &mut Connection, instruction: &EncoderInstruction) -> Res<u64> {
+        qdebug!([self], "insert instruction {:?}.", instruction);
+        if self.write_stream_blocked {
```

Comment:
Yes, I know all of this. there is couple of issues here. I stopped here because I need to fix transport, or think how to fix it or do something else.
My idea to fix this was to have a function on a stream Send_Blocked that will also mark stream as blocked, he stream still has some available space but still mark it as block . When flow control(connection or stream) is increase send event SendStreamWritable.

I still need to think about this a bit.

Reply:
to explain the function was called Send_Blocked because I wanted it to force sending BLOCK frame.

Reply:
Thinking about it, I will resolved this without transport dependencies, i.e. I will remove write_stream_blocked. We can add write_stream_blocked as a follow up, but I am not really sure that it is worth the complexity it is adding.

I will have an updated patch later this week, I am bit busy with other projects.